src/main/java/org/broadinstitute/hellbender/utils/GenotypeUtils.java

```diff
@@ -125,4 +139,50 @@ public static GenotypeCounts computeDiploidGenotypeCounts(final VariantContext v
     public static boolean genotypeIsUsableForAFCalculation(Genotype g) {
         return g.hasLikelihoods() || g.hasGQ() || g.getAlleles().stream().anyMatch(a -> a.isCalled() && a.isNonReference() && !a.isSymbolic());
     }
+
+    /**
+     * For a hom-ref, as long as we have GQ we can make a very accurate QUAL calculation
+     * since the hom-var likelihood should make a minuscule contribution
+     * @param g
+     * @param nAlleles
+     * @return log10 likelihoods
+     */
+    public static double[] makeApproximateLog10LikelihoodsFromGQ(Genotype g, int nAlleles) {
+        final int[] perSampleIndexesOfRelevantAlleles = new int[nAlleles];
+        Arrays.fill(perSampleIndexesOfRelevantAlleles, 1);
+        perSampleIndexesOfRelevantAlleles[0] = 0;  //ref still maps to ref
+        //use these values for diploid ref/ref, ref/alt, alt/alt likelihoods
+        final int gq = g.getGQ();
+        final int ploidy = g.getPloidy();
+        final int[] approxLikelihoods = {0, gq, PLOIDY_2_HOM_VAR_SCALE_FACTOR*gq};
+        //map likelihoods for any other alts to biallelic ref/alt likelihoods above
+        final int[] genotypeIndexMapByPloidy = GL_CALCS.getInstance(ploidy, nAlleles).genotypeIndexMap(perSampleIndexesOfRelevantAlleles, GL_CALCS); //probably horribly slow
```

Comment:
Took me a while to figure out what were you doing here.... also I'm a bit surprise it work.... would I work  if ```ploidy > 2```?
Can you add some large comment explaining the mapping for the sake of manteinabiliity?

Reply:
I see, I guess you might need to extend that ```approxLikelihoods``` array in size depending on the input ploidy.

Reply:
I tried making this ploidy-agnostic but couldn't come up with a clever way to fill out the approxLikelihoods for non-diploids.  I added ploidy checks, documentation, and a unit test.

Reply:
It seems to me that if the input genotype ploidy is not 2  (say 4) this would fail with a IoB exception. is not so?

Reply:
Darn, I forgot to push.  There should be a ploidy==2 check now.

Reply:
ok.